libpod/pod.go

```diff
@@ -0,0 +1,117 @@
+package libpod
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/docker/docker/pkg/stringid"
+)
+
+// Pod represents a group of containers that may share namespaces
+type Pod struct {
+	id   string
+	name string
+
+	containers map[string]*Container
+
+	valid bool
+	lock  sync.RWMutex
+}
+
+// ID retrieves the pod's ID
+func (p *Pod) ID() string {
+	return p.id
+}
+
+// Name retrieves the pod's name
+func (p *Pod) Name() string {
+	return p.name
```

Comment:
Are pod names changeable, Do we need locking here?

Reply:
I'd prefer to not allow pods or containers be renamed, but I'm not strongly opposed to adding this

Reply:
I would prefer that they not be renamed, but I don't know what is expected.  @mrunalp WDYT